### ISSUE TITLE
Request for comments: Should an IPV6 enabled libupnp fail initialization if there is no IPV6 available on the chosen interface?

### ISSUE THREAD

**Issue by mrjimenez** [pupnp/pupnp#193]
Hi folks,

As the title says, I am looking for opinions on how to fix (or not) the current behavior. But first, a little background information.

If you go to the project page today, you will notice that the branch master has a "failing" badge. This is because since we dropped the UpnpInit() method in favor of UpnpInit2(), the mac-os part of the automated tests stopped working. So this problem was always there, but just showed now. That is why I decided to release 1.14.0.

I don't have a Mac, so I had to deal with the virtual Github environment and I created a new branch to do some tests and found out that the mac-os virtual machine did not initialize the IPV6 stack in the en0 (Ethernet) interface. In spite of this, the library did initialize everything correctly until it tryed to create an UDP socket and tried to join IPV6 multicast, which is what triggered the failure.

Fixing the mac-os test failure is not a problem and has indeed been done in the auxiliary branch. But the question that remains is: should the library fail initialization or should it go on with IPV4? I am not the author of the original libupnp IPV6 code, so all I can do is read the code. It seems to me that lots of "ifdefs" were supposed to make the library code smaller, but that is not necessarily the case as the last patch I did on miniserver.c showed. Personally, I believe that if it does not imply a link time failure, being IPV6 enabled is not a problem even if it is not configured. Also, notice that IPV6 might be configured only in some interfaces, which makes things even more messy.

To summarize my thoughts:

1. I believe that the library initialization code should not fail if there is no IPV6 on the desired interface, as long as we can go on with IPV4.
1. I don't believe we need a configure time --enable-ipv6.

Since I can certainly be wrong in both cases, I ask for your comments.

Best regards,
Marcelo.

**Comment by G10h4ck** [comment]
I agree with both your points, still it is quite common to have compile time optional IPv6 support so if it doesn't require too much work to keep it I'd keep the compile time switch

**Comment by whyman** [comment]
The problem is that to get `UpnpInit2` you have to enable IPv6 anyway.

All consumers will now need `UpnpInit2`, so all will require the "optional" IPv6 support.

Therefore it is not actually optional any longer and should be enabled permanently.

Then the library should be made to not fail at runtime with pure v4.

So yes, to both both points from me.
